src/server/pfs/server/master_v2.go

```diff
@@ -21,7 +21,7 @@ const (
 func (d *driverV2) master(env *serviceenv.ServiceEnv, objClient obj.Client, db *gorm.DB) {
 	ctx := context.Background()
 	masterLock := dlock.NewDLock(d.etcdClient, path.Join(d.prefix, masterLockPath))
-	err := backoff.RetryNotify(func() error {
+	err := backoff.RetryUntilCancel(ctx, func() error {
```

Comment:
I don't think this is going to solve the problem. If the master context gets canceled, then the run function is going to return nil and we will exit the retry. We want the master goroutine to try and re-claim the master lock after it loses it. The retry until cancel should be in the polling function in the garbage collector.